staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -115,13 +115,13 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 		return false, 0
 
 	case io.ErrUnexpectedEOF:
-		klog.V(1).InfoS("Watch closed with unexpected EOF", "err", err)
+		klog.V(1).InfoS("Watch closed with unexpected EOF", "err", err, "resourceVersion", rw.lastResourceVersion)
 		return false, 0
 
 	default:
 		msg := "Watch failed"
 		if net.IsProbableEOF(err) || net.IsTimeout(err) {
-			klog.V(5).InfoS(msg, "err", err)
+			klog.V(5).InfoS(msg, "err", err, "resourceVersion", rw.lastResourceVersion)
```

Comment:
A few lines down, on 129 there's an entry which could use the `resourceVersion` as well

```
klog.ErrorS(err, msg)
```